services/collateral.go

```diff
@@ -28,25 +29,39 @@ func (s *Service) GetCollateralPrice(blockNumber *big.Int, collateralType common
 }
 
 func (s *Service) RetrieveCollateralWithdrawnLimit(limit uint64) ([]*models.CollateralWithdrawn, error) {
-	iterations, last, err := s.getIterationsForLimitQueryCore(limit)
+	return s.RetrieveCollateralWithdrawn(0, 0, limit)
+}
+
+func (s *Service) RetrieveCollateralWithdrawn(fromBlock uint64, toBlock uint64, limit uint64) ([]*models.CollateralWithdrawn, error) {
+	iterations, lastBlock, err := s.getIterationsForLimitQueryCore(limit)
 	if err != nil {
 		return nil, err
 	}
 
+	if fromBlock == 0 {
+		fromBlock = s.coreFirstBlock
+	}
+
 	var withdraws []*models.CollateralWithdrawn
 
 	logger.Log().WithField("layer", "Service-RetrieveCollateralWithdrawnLimit").Infof(
```

Comment:
Service-RetrieveCollateralWithdrawn instead of
"Service-RetrieveCollateralWithdrawnLimit